vividus-allure-adaptor/src/main/java/org/vividus/report/allure/AllurePluginsProvider.java

```diff
@@ -0,0 +1,39 @@
+/*
+ * Copyright 2019-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.vividus.report.allure;
+
+import java.util.List;
+
+import javax.inject.Inject;
+
+import io.qameta.allure.core.Plugin;
+
+public class AllurePluginsProvider
```

Comment:
do we need this list container? can't plugins list be injected into `AllureReportGenerator` constructor?

Reply:
This provider is needed to have ability to manage allure plugins on runtime.

Reply:
could you please elaborate on this?

Reply:
The problem here is that without direct defining of list of beans spring generates different instances of lists for all places where this beans should be used. As the result you will not have access to list instance in AllureReportGenerator, so there will not be a possibility to dynamically change this list.

Reply:
ok, that's clear, but why do you need to change the list dynamically?